src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/ProcessInitialDataSetVersionFunction.cs

```diff
@@ -58,6 +85,67 @@ public class ProcessInitialDataSetVersionFunction(PublicDataDbContext publicData
         }
     }
 
+    [Function(nameof(ImportMetadata))]
+    public async Task ImportMetadata(
+        [ActivityTrigger] Guid dataSetVersionId,
+        Guid instanceId,
+        CancellationToken cancellationToken)
+    {
+        var dataSetVersion = await GetDataSetVersion(
+            dataSetVersionId: dataSetVersionId,
+            instanceId: instanceId,
+            cancellationToken);
+
+        var dataSetVersionImport = dataSetVersion.Imports.Single(i => i.InstanceId == instanceId);
+
+        dataSetVersionImport.Stage = DataSetVersionImportStage.ImportingMetadata;
+        await publicDataDbContext.SaveChangesAsync(cancellationToken);
+
+        await dataSetMetaService.CreateDataSetVersionMeta(dataSetVersion, cancellationToken);
+    }
+
+    [Function(nameof(ImportData))]
+    public async Task ImportData(
+        [ActivityTrigger] Guid dataSetVersionId,
+        Guid instanceId,
+        CancellationToken cancellationToken)
+    {
+        var dataSetVersion = await GetDataSetVersion(
+            dataSetVersionId: dataSetVersionId,
+            instanceId: instanceId,
+            cancellationToken);
+
+        var dataSetVersionImport = dataSetVersion.Imports.Single(i => i.InstanceId == instanceId);
+
+        dataSetVersionImport.Stage = DataSetVersionImportStage.ImportingData;
+        await publicDataDbContext.SaveChangesAsync(cancellationToken);
+
+        await using var duckDbConnection =
+            DuckDbConnection.CreateFileConnection(dataSetVersionPathResolver.DuckDbPath(dataSetVersion));
+        duckDbConnection.Open();
```

Comment:
I've done this now. The only reason why I didn't do it before was because it's not consistent with the other repository classes which have methods that accept the connection, so that when processing the meta data we reuse the same connection which is opened once.